File: mlkit-pose-detection/src/main/java/com/king/mlkit/vision/pose/PoseCameraScanActivity.java
package com.king.mlkit.vision.pose;

import androidx.annotation.Nullable;

import com.google.mlkit.vision.pose.Pose;
import com.king.mlkit.vision.camera.BaseCameraScanActivity;
import com.king.mlkit.vision.camera.analyze.Analyzer;
import com.king.mlkit.vision.pose.analyze.PoseDetectionAnalyzer;

/**
 * @author <a href="mailto:jenly1314@gmail.com">Jenly</a>
 */
public abstract class PoseCameraScanActivity extends BaseCameraScanActivity<Pose> {

    @Nullable
    @Override
    public Analyzer<Pose> createAnalyzer() {
        return new PoseDetectionAnalyzer();
    }
}

File: mlkit-pose-detection/src/main/java/com/king/mlkit/vision/pose/PoseCameraScanActivity.kt
package com.king.mlkit.vision.pose

import com.google.mlkit.vision.pose.Pose
import com.king.camera.scan.BaseCameraScanActivity
import com.king.camera.scan.analyze.Analyzer
import com.king.mlkit.vision.pose.analyze.PoseDetectionAnalyzer

abstract class PoseCameraScanActivity : BaseCameraScanActivity<Pose>() {
    override fun createAnalyzer(): Analyzer<Pose>? = PoseDetectionAnalyzer()
}
